hikari/api/event_manager.py

```diff
@@ -113,10 +110,10 @@ def open(self) -> None:
 
     @abc.abstractmethod
     def filter(
-        self: _T,
+        self,
         *predicates: typing.Union[typing.Tuple[str, typing.Any], typing.Callable[[EventT], bool]],
         **attrs: typing.Any,
-    ) -> _T:
+    ) -> EventStream[EventT]:
```

Comment:
Why was this changed

Reply:
Removes an unnecessary typevar

Reply:
So for no reason?

Reply:
As the pr says, this is just to cleanup typehints

Reply:
well if you want to go and change all the places which return Self and are typed with typevars rather than just the actual type (since this pattern works better with inheritance) because apparently that's messier(?) then go ahead ig

Reply:
also in the future this'll pattern will be replaced by typing.Self but that's way far off

Reply:
But yeah changing this could cause issues for anybody who's extending this interface for cases like
```py
class ExtendedEventStream(EventStream[EventT]):
    @abc.abstractmethod
    async def foo(self) -> int:
        raise NotImplementedError
 
def do_thing(stream: ExtendedEventStream[MemberCreateEvent]) -> int:
    return await stream.filter(lambda e: e.user.id in {...}).foo()
```

since you're removing the guarantee that the return type will always be Self

Reply:
Didnt realize at the time how dumb this was to remove. Fixed :+1: